app/controllers/register_controller.rb

```diff
@@ -33,19 +34,27 @@ def edit
   end
 
   def confirm
-    return true if params[:data_confirmed]
-    @register = get_register(params[:register])
-
-    @current_register_record = OpenRegister.record(params[:register].downcase,
-                        params[params[:register].downcase.to_sym],
-                        :beta)
-
-    if @current_register_record != nil
-      @current_register_record = convert_register_json(@current_register_record)
+    register_name = params[:register].downcase
+    field_definitions = @registers_client.get_register(params[:register], 'beta').get_field_definitions
+    validation_result = @data_validator.get_form_errors(params, field_definitions, register_name)
+    if validation_result.messages.present?
+      validation_result.messages.each { |k,v| flash[k] = v.join(', ') }
+      @register = get_register(register_name)
+      @form = JSON.parse(params.to_json)
+      render :new
+    else
+      return true if params[:data_confirmed]
+      @register = get_register(params[:register])
+      @current_register_record = OpenRegister.record(register_name,
+                          params[register_name.to_sym],
+                          :beta)
+
+      if @current_register_record != nil
```

Comment:
What happens if it is `nil`? Will rendering `:confirm` explode with missing fields?

Reply:
This needs changing to just `if @current_register_record`

Reply:
actually this isn't doing anything now 🤔